src/Plutarch/Test/QuickCheck/Instances.hs

```diff
@@ -0,0 +1,974 @@
+pliftT (TestableTerm x) = plift x
+
+pconstantT :: (PLift p, PLifted p ~ h) => h -> TestableTerm p
+pconstantT h = TestableTerm $ pconstant h
+
+pconT :: PlutusType p => (forall s. p s) -> TestableTerm p
+pconT p = TestableTerm $ pcon p
+
+pmatchT :: PlutusType p => TestableTerm p -> (forall s. p s -> Term s b) -> TestableTerm b
+pmatchT (TestableTerm p) f = TestableTerm $ pmatch p f
+
+-- | @since 2.0.0
+instance (PArbitrary p, PIsData p) => PArbitrary (PAsData p) where
+    parbitrary = do
+        (TestableTerm x) <- parbitrary
+        return $ TestableTerm $ pdata x
+    pshrink = fmap pdataT . shrink . pfromDataT
+
+instance (PCoArbitrary p, PIsData p) => PCoArbitrary (PAsData p) where
+    pcoarbitrary (pfromDataT -> x) = pcoarbitrary x
+
+instance Function (TestableTerm PInteger) where
+    function = functionMap pliftT pconstantT
+
+-- | @since 2.0.0
+instance PArbitrary PInteger where
+    pshrink = fmap pconstantT . shrink . pliftT
+
+instance PCoArbitrary PInteger where
+    pcoarbitrary (pliftT -> x) = coarbitrary x
+
+-- | @since 2.0.0
+instance PArbitrary PBool where
+    pshrink = fmap pconstantT . shrink . pliftT
+
+instance PCoArbitrary PBool where
+    pcoarbitrary (pliftT -> x) = coarbitrary x
+
+-- | @since 2.0.0
+instance PArbitrary PUnit where
+    pshrink = fmap pconstantT . shrink . pliftT
+
+instance PCoArbitrary PUnit where
+    pcoarbitrary (pliftT -> x) = coarbitrary x
+
+genByteStringUpto :: Int -> Gen ByteString
+genByteStringUpto m = sized go
+  where
+    go :: Int -> Gen ByteString
+    go s = chooseInt (0, min m s) >>= genByteString
+
+genByteString :: Int -> Gen ByteString
+genByteString l = Exts.fromList <$> vectorOf l arbitrary
+
+shrinkByteString :: ByteString -> [ByteString]
+shrinkByteString bs = do
+    xs' <- shrink . Exts.toList $ bs
+    pure . Exts.fromList $ xs'
+
+-- | @since 2.0.0
+instance PArbitrary PByteString where
+    parbitrary = do
+        len <- chooseInt (0, 64)
+        bs <- genByteString len
+        return $ TestableTerm $ pconstant bs
+
+    pshrink = fmap pconstantT . shrinkByteString . pliftT
+
+-- | @since 2.0.0
+instance PCoArbitrary PByteString where
+    pcoarbitrary = coarbitrary . sum . Exts.toList . pliftT
+
+-- | @since 2.0.0
+instance PArbitrary PPositive where
+    parbitrary = do
+        (TestableTerm x) <- parbitrary
+        return $ TestableTerm $ ptryPositive #$ pif (0 #< x) x (negate (x + 1))
+
+-- | @since 2.0.0
+instance PCoArbitrary PPositive where
+    pcoarbitrary = pcoarbitrary . liftTestable pto
+
+-- | @since 2.0.0
+instance PArbitrary PRational where
+    parbitrary = do
+        (TestableTerm x) <- parbitrary
+        (TestableTerm y) <- parbitrary
+        return $ TestableTerm $ pcon $ PRational x y
+
+    pshrink (TestableTerm x) =
+        [ TestableTerm $ pcon $ PRational a b
+        | (TestableTerm a) <- shrink (TestableTerm $ pnumerator # x)
+        , (TestableTerm b) <- shrink (TestableTerm $ pdenominator # x)
+        ]
```

Comment:
This shrinker is a tad wonky. If I understand correctly, `PRational`s are normalized. This means that this shrinker could inadvertently make a fraction _bigger_.

I would recommend numerator-only shrinking instead.